### ISSUE TITLE
Sort and map filters aren't resilient to nulls

### ISSUE THREAD

**Issue by tpluscode** [dotliquid/dotliquid#269]
### Dotliquid version

2.0.183

### Expected behavior

When `sort` and `map` filter is passed null it should silently ignore it.

### Actual behavior

Depending on the setup `ArgumentNullException` or `NullReferenceException` is thrown

### Steps to reproduce the Problem (you can add files)

``` c#
var tmpl = DotLiquid.Template.Parse("x: '{{ test.deep | sort }}'");
var model = Hash.FromDictionary(new Dictionary<string, object>
{
   { "test", null }
});

tmpl.Render(model);
```

### Workaround

Here's a simple fix we applied in our project

``` c#
public static class ResilientStandardFilters
{
    public static IEnumerable Sort(object input, string property = null)
    {
        if (input == null)
        {
            return new object[0];
        }

        return StandardFilters.Sort(input, property);
    }

    public static IEnumerable Map(IEnumerable input, string property)
    {
        if (input == null)
        {
            return new object[0];
        }

        return StandardFilters.Map(input, property);
    }
}
```

**Comment by nytmo** [comment]
Consider changing the Map's filter first row from:
`List<object> ary = input.Cast<object>().ToList();`
to:
`List<object> ary = input.Cast<object>().Where(o => o != null).ToList();`

**Comment by tpluscode** [comment]
That would fix null elements of the `input` collection, which I don't think is broken by the way. The problem is when `input` itself in null.

**Comment by nytmo** [comment]
Tested null elements and it's a problem, as later in Map method there is this:
`if (ary.All(o => o.RespondTo(property)))`
which leads to `NullReferenceException` when `o is null`

**Comment by tpluscode** [comment]
I'd have to try again but from what you're saying there are in fact two possibilities for improvement?

**Comment by nytmo** [comment]
Yes, there are two bugs concerning null handling. And I have reported another issue concerning naming conventions of the property name used in map filter. But it's kind a different story.

**Comment by nytmo** [comment]
Resolved by PR #294 . The issue may be closed, I guess.
